porcupine/settings.py

```diff
@@ -578,6 +578,47 @@ def add_entry(
     return entry
 
 
+def add_checkbutton(option_name: str, **checkbutton_kwargs: Any) -> ttk.Entry:
+    """Add a :class:`tkinter.ttk.Checkbutton` to the setting dialog.
+
+    All ``**checkbutton_kwargs`` go to :class:`tkinter.ttk.Entry`.
+    You can do this, for example::
+
+        from porcupine import settings
+
+        def do_something() -> None:
+            if settings.get("foobar", bool):
+                print("Foobar enabled")
+            else:
+                print("Foobar disabled")
+
+        def setup() -> None:
+            settings.add_option("foobar", False, bool)  # False is default value
```

Comment:
It uses `type(False)` if `bool` is not given:

```suggestion
            settings.add_option("foobar", False)  # False is default value
```